GDJS/Runtime/variable.ts

```diff
@@ -211,73 +285,205 @@ namespace gdjs {
      * @param newValue The new string to be set
      */
     setString(newValue: string): void {
-      this._str = newValue;
-      this._numberDirty = true;
-      this._stringDirty = false;
+      this._type = 'string';
+      this._str = '' + newValue;
+    }
+
+    /**
+     * Get the value of the variable, considered as a boolean
+     * @return {boolean}
+     */
+    getAsBoolean(): boolean {
+      if (this._type !== 'boolean') {
+        if (this._type === 'number') return this._value !== 0;
+        else if (this._type === 'string')
+          return this._str !== '0' && this._str !== '' && this._str !== 'false';
+        else return this.getChildrenCount() !== 0;
+      }
+
+      return this._bool;
+    }
+
+    /**
+     * Change the value of the variable, considered as a boolean
+     * @param {boolean} newValue
+     */
+    setBoolean(newValue: boolean) {
+      this._type = 'boolean';
+      this._bool = !!newValue;
+    }
+
+    /**
+     * Sets the primitive value using the setter of the current type.
+     * @param {any} newValue
+     */
+    setValue(newValue: string | float | boolean) {
+      if (this._type === 'string') this.setString(newValue as string);
+      else if (this._type === 'number') this.setNumber(newValue as float);
+      else if (this._type === 'boolean') this.setBoolean(newValue as boolean);
+    }
+
+    /**
+     * Gets the primitive value using the getter of the current type.
+     * @param {any} newValue
```

Comment:
Remove this

Reply:
What should be removed? The function? The type in the parameter? The whole parameter tag?

Reply:
The @param is not associated to any param.

Reply:
So the whole param tag